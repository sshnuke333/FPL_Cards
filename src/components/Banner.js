import React from 'react';
import {
    StyledDiv,
    BannerText,
    StyledLink,
    HideBanner,
    Close,
} from './Banner.styles';
import { toggleDisplay } from '../helpers/toggleDisplay';

export const Banner = () => {
    return (
        <StyledDiv id="alert" aria-label="alert">
            <BannerText>
                This is not a licensed Premier League app please read the
                complete <StyledLink to="disclaimer">disclaimer</StyledLink>{' '}
            </BannerText>
            <HideBanner
                aria-hidden="false"
                onClick={(e) =>
                    toggleDisplay(e, 'alert', 'none', 'aria-hidden')
                }
            >
                <Close verticalAlign="0" />
            </HideBanner>
        </StyledDiv>
    );
};
