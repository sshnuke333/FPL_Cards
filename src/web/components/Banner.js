import React from 'react';
import {
    StyledDiv,
    BannerText,
    StyledLink,
    HideBanner,
    Close,
} from './Banner.styles';
import { expandTarget, hideDisplay } from '../helpers/display';

export const Banner = () => {
    return (
        <StyledDiv id="alert" aria-label="alert">
            <BannerText>
                This is not a licensed Premier League app please read the
                complete{' '}
                <StyledLink
                    to="disclaimer"
                    onClick={() => hideDisplay('alert')}
                >
                    disclaimer
                </StyledLink>{' '}
            </BannerText>
            <HideBanner
                aria-label="close"
                aria-hidden="false"
                onClick={(e) => expandTarget(e, 'alert', 'none', 'aria-hidden')}
            >
                <Close verticalAlign="0" />
            </HideBanner>
        </StyledDiv>
    );
};
