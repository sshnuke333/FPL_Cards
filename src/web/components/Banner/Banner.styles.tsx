import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { XCircleIcon } from '@primer/octicons-react';

const textStyle = `
    font-size: 1.5rem;
    color: #3d003f;
    @media only screen and (max-width: 996px) {
        font-size: 1.25rem;
    }
    @media only screen and (max-width: 768px) {
        font-size: 1rem;
    }`;

export const StyledDiv = styled.div`
    height: 4rem;
    display: grid;
    place-items: center;
    grid-template-columns: auto 3rem;
    background: linear-gradient(45deg, #fe6900, #dad400);
    font-size: 2rem;
    @media only screen and (max-width: 1200px) {
        height: 4.5rem;
    }
`;
export const BannerText = styled.span`
    padding: 0.5rem 1rem;
    ${textStyle}
`;

export const StyledLink = styled(Link)`
    ${textStyle}
    text-decoration: underline 3px #3d003f;
    &:hover {
        color: #fe6900;
    }
`;

export const HideBanner = styled.button`
    width: 2rem;
    height: 2rem;
    margin: 0.5rem;
    background-color: transparent;
    cursor: pointer;
`;
export const Close = styled(XCircleIcon)`
    width: 1.5rem;
    height: 1.5rem;
    pointer-events: none;
    vertical-align: 0;
`;
