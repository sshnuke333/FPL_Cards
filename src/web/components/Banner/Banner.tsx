import { expandTarget, hideDisplay } from '../../helpers/display';
import {
    StyledDiv,
    BannerText,
    StyledLink,
    HideBanner,
    Close,
} from './Banner.styles';

const Banner = (): JSX.Element => (
    <StyledDiv id="alert" aria-label="alert">
        <BannerText>
            This is not a licensed Premier League app please read the complete{' '}
            <StyledLink to="disclaimer" onClick={() => hideDisplay('alert')}>
                disclaimer
            </StyledLink>{' '}
        </BannerText>
        <HideBanner
            aria-label="close"
            aria-hidden="false"
            onClick={(e) => expandTarget(e, 'alert', 'none', 'aria-hidden')}
        >
            <Close />
        </HideBanner>
    </StyledDiv>
);

export default Banner;
