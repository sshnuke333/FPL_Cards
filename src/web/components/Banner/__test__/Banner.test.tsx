import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';
import Banner from '../Banner';

describe('Banner', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <Router>
                <Banner />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });
    beforeEach(() => {
        render(
            <Router>
                <Banner />
            </Router>
        );
    });
    let alert: HTMLElement;
    let closeButton: HTMLElement;
    it('renders a alert banner', () => {
        expect(screen.getByLabelText(/alert/i)).toBeInTheDocument();
    });
    it('renders disclaimer link correctly', () => {
        const disclaimerLink = screen.getByRole('link', {
            name: /disclaimer/i,
        });
        expect(disclaimerLink).toBeInTheDocument();
        expect(disclaimerLink).toBeVisible();
        userEvent.click(disclaimerLink);
        expect(disclaimerLink).not.toBeVisible();
    });
    it('renders a clickable button inside alert correctly', () => {
        alert = screen.getByLabelText(/alert/i);
        closeButton = within(alert).getByRole('button', {
            hidden: false,
        });
        expect(closeButton).toBeVisible();
        expect(closeButton).toBeInTheDocument();
        expect(closeButton).not.toBeDisabled();
        expect(closeButton).toHaveAttribute('aria-label', 'close');
        userEvent.click(closeButton);
        expect(alert).not.toBeVisible();
    });
});
