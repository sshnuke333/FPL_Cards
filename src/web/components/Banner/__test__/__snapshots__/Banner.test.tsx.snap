// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="alert"
    class="sc-bcXHqe hWTUaF"
    id="alert"
  >
    <span
      class="sc-gswNZR eqEKMx"
    >
      This is not a licensed Premier League app please read the complete 
      <a
        class="sc-dkrFOg hbCuPf"
        href="/disclaimer"
      >
        disclaimer
      </a>
       
    </span>
    <button
      aria-hidden="false"
      aria-label="close"
      class="sc-hLBbgP iNmBsD"
    >
      <svg
        aria-hidden="true"
        class="sc-eDvSVe fIEffj"
        fill="currentColor"
        focusable="false"
        height="16"
        role="img"
        style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M3.404 12.596a6.5 6.5 0 119.192-9.192 6.5 6.5 0 01-9.192 9.192zM2.344 2.343a8 8 0 1011.313 11.314A8 8 0 002.343 2.343zM6.03 4.97a.75.75 0 00-1.06 1.06L6.94 8 4.97 9.97a.75.75 0 101.06 1.06L8 9.06l1.97 1.97a.75.75 0 101.06-1.06L9.06 8l1.97-1.97a.75.75 0 10-1.06-1.06L8 6.94 6.03 4.97z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
